test/unit/picks-handler.spec.ts

```diff
@@ -121,3 +122,124 @@ describe("when getting the stats of a pick", () => {
     })
   })
 })
+
+describe("when getting the picks for an item", () => {
+  let url: URL
+  let getPicksByItemIdMock: jest.Mock
+  let request: HandlerContextWithPath<"picks", "/v1/picks/:itemId">["request"]
+  let params: HandlerContextWithPath<"picks", "/v1/picks/:itemId">["params"]
+  let userAddress: string
+  let anotherUserAddress: string
+  let dbPicksByItemId: DBGetFilteredPicksWithCount[]
+  let picks: Pick<TPick, "userAddress">[]
+
+  beforeEach(() => {
+    itemId = "item-id"
+    getPicksByItemIdMock = jest.fn()
+    components = {
+      picks: createTestPicksComponent({ getPicksByItemId: getPicksByItemIdMock }),
+    }
+    request = {} as HandlerContextWithPath<"lists", "/v1/lists/:id/picks">["request"]
+    url = new URL(`http://localhost/v1/lists/${itemId}/picks`)
+    params = { itemId }
+    userAddress = "0x687abb534BD927284F84b03d43f33dF0E5C91D21"
+    anotherUserAddress = "0x45abb534BD927284F84b03d43f33dF0E5C91C21f"
+
+    dbPicksByItemId = [
+      {
+        item_id: "1",
+        user_address: userAddress,
+        list_id: "e96df126-f5bf-4311-94d8-6e261f368bb2",
+        created_at: new Date(),
+        picks_count: 2,
+      },
+      {
+        item_id: "1",
+        user_address: anotherUserAddress,
+        list_id: "f96df126-f5bf-4311-94d8-6e261f368bb4",
+        created_at: new Date(),
+        picks_count: 2,
+      },
+    ]
+  })
+
+  describe("and the process to get the picks fails", () => {
+    let error: Error
+
+    beforeEach(() => {
+      error = new Error("anError")
+      getPicksByItemIdMock.mockRejectedValueOnce(error)
+    })
+
+    it("should propagate the error", () => {
+      return expect(getPicksByItemIdHandler({ params, components, url, request })).rejects.toEqual(error)
+    })
+  })
+
+  describe("and the process to get the picks is successful", () => {
+    describe("when not using pagination parameters", () => {
+      beforeEach(() => {
+        picks = [{ userAddress }, { userAddress: anotherUserAddress }]
+        getPicksByItemIdMock.mockResolvedValueOnce(dbPicksByItemId)
+      })
+
+      it("should return a response with an ok status code and the picks using the default values of limit and page", () => {
+        return expect(getPicksByItemIdHandler({ url, components, request, params })).resolves.toEqual({
+          status: StatusCode.OK,
+          body: {
+            ok: true,
+            data: {
+              results: picks,
+              total: 2,
+              page: 0,
+              pages: 1,
+              limit: 100,
+            },
+          },
+        })
+      })
+    })
+
+    describe("when using the pagination parameters", () => {
+      it("should return an array with the first pick when the limit is 1 and the page is 0", () => {
```

Comment:
We usually use the `describes` to describe the context, the `beforeEach` to program the described context, the `it` to describe what's expected from the test under the describe context and the scope of the `it` to program the expectation. Would you mind adding the describe and the `beforeEach` for each of the its?
```suggestion
    describe("when using the pagination parameters", () => {
      describe("when the limit is 1 and the page is 0", () => {
         beforeEach(() => {
            url = new URL(`http://localhost/v1/lists/${itemId}/picks?limit=1&page=0`)
            picks = [{ userAddress }]
            getPicksByItemIdMock.mockResolvedValueOnce([dbPicksByItemId[0]])
         })
         it("should return an array with the first pick, () => {
```